### PR TITLE
Problem with overwrite node template if it's more than one menu

### PR DESCRIPTION
**Pull request by dawiddczaja**
Internal ticket DEV-37921

### CODE REVIEW

**Comment by DariuszMaciejewski** [comment]
@dawiddczaja could you explain why we can't use `$this->_template`?

**Comment by dawiddczaja** [comment]
@DariuszMaciejewski Yes sure, when node (e.g. category) is rendered first time `$this->_template` was set in property like  `protected $_template = 'menu/node_type/category.phtml';` then it's go to the function 

https://github.com/SnowdogApps/magento2-menu/blob/master/Block/NodeType/AbstractNode.php#L138

And if template is different then default,  function ` $this->setTemplate($template);` assigned to variable `$this->_template` new value so next time when same node try render it's paste argument to function `getMenuTemplate` it's not default template.

This behavior corrupts function https://github.com/SnowdogApps/magento2-menu/blob/master/Model/TemplateResolver.php#L36

File: Block/NodeType/AbstractNode.php [hunk 1]
```diff
@@ -131,7 +136,7 @@ public function isViewAllLinkAllowed()
     {
         return $this->viewAllLink;
     }
-
+
```
Comment (DariuszMaciejewski) [inline Block/NodeType/AbstractNode.php]: remove extra whitespace

File: Block/NodeType/AbstractNode.php [hunk 2]
```diff
@@ -21,6 +21,11 @@
     const NAME_CODE = 'node_name';
     const CLASSES_CODE = 'node_classes';
 
+    /**
+     * @var string
+     */
+    protected $defaultTemplate = '';
```
Comment (maciekkossuda) [inline Block/NodeType/AbstractNode.php]: Empty string is not the best solution to initialize var.

**Pull request merged**